multipool-tracker/src/multipool_storage/expiry.rs

```diff
@@ -1,15 +1,27 @@
+use super::errors::MultipoolErrors;
 use std::time::{SystemTime, UNIX_EPOCH};
 
 #[derive(Clone, Debug)]
 pub struct MayBeExpired<V>(V, u64);
 
+impl<V, E> MayBeExpired<Result<V, E>> {
+    pub fn transpose(self) -> Result<MayBeExpired<V>, E> {
```

Comment:
This pr can potentially remove this function as we want to return error in all option places that are lacking some storage data. Function was used only in these places.